### PR TITLE
Revert symbol-annotation bump

### PR DESCRIPTION
**Pull request by jglick**
Reverts #5265. Noticed in https://github.com/jenkinsci/file-parameters-plugin/pull/27 that this caused a build failure

```
[WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.jenkins-ci:symbol-annotation:1.20 paths to dependency are:
+-io.jenkins.plugins:file-parameters:98.728b4f423129
  +-org.jenkins-ci.plugins:structs:1.20
    +-org.jenkins-ci:symbol-annotation:1.20
and
+-io.jenkins.plugins:file-parameters:98.728b4f423129
  +-org.jenkins-ci.main:jenkins-core:2.281-rc30899.58174df06127
    +-org.jenkins-ci:symbol-annotation:1.20 (managed) <-- org.jenkins-ci:symbol-annotation:1.21
]
```

This is the most straightforward way to fix the immediate regression. There are probably other workarounds, though _not_ using `optional`—we do want it bundled in the WAR! Really the root of the evil is that we have this JAR which is both a core library dep _and_ bundled in a plugin. I do not recall why things were set up this way exactly; probably to avoid needing to use new core deps in plugins, at a time when that seemed really scary.

## Proposed changelog entries

* Developer: Downgrade the bundled symbol-annotation library version to prevent build failures in plugins

### CODE REVIEW

File: core/pom.xml
Comment (timja) [inline core/pom.xml:521]: why not just bump the version of structs? you just need to use bom version 24.

I noticed this in https://github.com/jenkinsci/plot-plugin/pull/65, but bumping bom fixed it

**Comment by timja** [comment]
> This is the most straightforward way to fix the immediate regression. There are probably other workarounds, though not using optional—we do want it bundled in the WAR! Really the root of the evil is that we have this JAR which is both a core library dep and bundled in a plugin. I do not recall why things were set up this way exactly; probably to avoid needing to use new core deps in plugins, at a time when that seemed really scary.

Could we remove it from the structs plugin?

File: core/pom.xml
Comment (jglick) [reply core/pom.xml:521]: The particular version does not matter; it has not changed since 1.0 AFAIK. I just do not want it to be constantly bumped up here.

**Comment by jglick** [comment]
> Could we remove it from the structs plugin?

Yes, the proper fix would be to split to an independent repo, remove it as a dep from `structs`, and just expect it to be picked up from `jenkins-core` like any other bundled lib (managed in `jenkins-bom`). This patch is more of a hotfix.

File: core/pom.xml
Comment (timja) [reply core/pom.xml:521]: Which one is actually used at runtime? the structs one or the Jenkins core one?

Comment (jglick) [reply core/pom.xml:521]: The Jenkins core one. Which does not really matter because

```bash
git diff structs-parent-1.1 -- annotation/src/main
```

```diff
diff --git annotation/src/main/java/org/jenkinsci/Symbol.java annotation/src/main/java/org/jenkinsci/Symbol.java
index abd45ef..7f007e2 100644
--- annotation/src/main/java/org/jenkinsci/Symbol.java
+++ annotation/src/main/java/org/jenkinsci/Symbol.java
@@ -33,6 +33,10 @@ import static java.lang.annotation.RetentionPolicy.*;
  * The first one is used as the primary identifier for reverse-mapping.
  *
  * <p>
+ * Generally symbols should be named after their display names, so that users see
+ * association between what they see in web UI and symbols they see in code.
+ *
+ * <p>
  * To look up a component by its symbol, see the documentation of the symbol plugin.
  */
 @Indexed
```

Comment (oleg-nenashev) [reply core/pom.xml:521]: The last code change was in 1.0, so it does not matter much: https://github.com/jenkinsci/structs-plugin/commit/2efc80dc826685a43f717595d64307ad7f474a44#diff-94e7f2d13ab308fa2df75a0097ab64be6d274e2a8233741ca4c45f1195602c71 . The produced bytecode is likely to be different between versions though.

**Review by MarkEWaite — APPROVED** [review]
When I reviewed changes in the sources that contribute to symbol-annotation.jar, they had not changed since at least version 1.3 and it may have been there were no changes since even before that.

No problem for me if we hold this at version 1.1.

**Comment by MarkEWaite** [comment]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

I've seen the same problem in platformlabeler plugin builds that @jglick reported in https://github.com/jenkinsci/file-parameters-plugin/pull/27

File: core/pom.xml
Comment (timja) [reply core/pom.xml:521]: I still don't understand why structs doesn't just pickup the jenkins core one?

Shouldn't we just remove the dependency there, what if we did want to update the library?

**Comment by daniel-beck** [comment]
Mark commented but didn't label, doing so now.

**Comment by MarkEWaite** [comment]
@daniel-beck I think my listing it as ready to merge was premature.  I believe we still need a second approval before it is merged.

**Review by daniel-beck — APPROVED** [review]
Reasonable short-term fix at least

**Review by oleg-nenashev — APPROVED** [review]
Should be good to go

File: core/pom.xml
Comment (jglick) [reply core/pom.xml:521]: > why structs doesn't just pickup the jenkins core one

It could, if the annotation component were split to its own repo with its own lifecycle.

Comment (basil) [reply core/pom.xml:521]: > It could, if the annotation component were split to its own repo with its own lifecycle.

@timja @jglick I extracted this to a separate repository here: https://github.com/basil/lib-symbol-annotation

What are the next steps? Get this hosted under the `jenkinsci` GitHub organization, add myself to RPU and cut a release, and then update core to use the new release and `structs` plugin to use the version from core? Can the former be sorted casually or shall I file a `HOSTING` request?

Comment (timja) [reply core/pom.xml:521]: Can do you a hosting request please.

Comment (basil) [reply core/pom.xml:521]: Done: [HOSTING-1155](https://issues.jenkins.io/browse/HOSTING-1155)

**Pull request merged**
